sdk/internal/recording/recording.go

```diff
@@ -655,6 +655,33 @@ func getRootCas(t *testing.T) (*x509.CertPool, error) {
 	return rootCAs, nil
 }
 
+type RecordingHTTPClient struct {
+	defaultClient *http.Client
+	options       RecordingOptions
+	t             *testing.T
+}
+
+func (c RecordingHTTPClient) Do(req *http.Request) (*http.Response, error) {
+	c.options.ReplaceAuthority(c.t, req)
+	return c.defaultClient.Do(req)
+}
+
+func GetRoutingHTTPClient(t *testing.T, options *RecordingOptions) (*RecordingHTTPClient, error) {
```

Comment:
Is this basically `NewRecordingHTTPClient`? It looks like it fits that pattern (I don't see 'Routing' anywhere in here so it feels like that nomenclature is outdated)

Reply:
Changed the name